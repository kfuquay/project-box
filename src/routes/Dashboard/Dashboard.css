.list-header {
  text-align: center;
  color: whitesmoke;
  font-size: 4vw;
  margin: 50px auto;
  text-transform: uppercase;
  letter-spacing: 0.09em;
  font-weight: 300;
}

.list-projecttiles {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-evenly;
  width: 80%;
  margin: 0 auto;
  max-width: 1200px;
}

.search-container, 
.switch-container {
	width: 80%;
	margin: 8px auto;
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
}

.search-container {
	margin-bottom: 50px;
}

#search {
	padding: 15px;
	border: none;
	max-width: 500px;
}

.screen-reader-label {
	border: 0;
    clip: rect(1px, 1px, 1px, 1px);
    clip-path: inset(50%);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    position: absolute !important;
    width: 1px;
    word-wrap: normal !important;
}

.switch {
	position: relative;
    display: block;
	width: 100px;
	height: 30px;
	padding: 3px;
	margin: 0 10px 10px 0;
	background: linear-gradient(to bottom, #eeeeee, #FFFFFF 25px);
	background-image: -webkit-linear-gradient(top, #eeeeee, #FFFFFF 25px);
	border-radius: 18px;
	box-shadow: inset 0 -1px white, inset 0 1px 1px rgba(0, 0, 0, 0.05);
	cursor: pointer;
	box-sizing:content-box;
}
.switch-input {
	position: absolute;
	top: 0;
	left: 0;
	opacity: 0;
	box-sizing:content-box;
}
.switch-label {
	position: relative;
	display: block;
	height: inherit;
    font-size: 12px;
    letter-spacing: .09em;
	text-transform: uppercase;
	background: #eceeef;
	border-radius: inherit;
	box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.12), inset 0 0 2px rgba(0, 0, 0, 0.15);
	box-sizing:content-box;
}
.switch-label:before, .switch-label:after {
	position: absolute;
	top: 50%;
	margin-top: -.5em;
	line-height: 1;
	-webkit-transition: inherit;
	-moz-transition: inherit;
	-o-transition: inherit;
	transition: inherit;
	box-sizing:content-box;
}
.switch-label:before {
	content: attr(data-off);
	right: 11px;
	color: #aaaaaa;
	text-shadow: 0 1px rgba(255, 255, 255, 0.5);
}
.switch-label:after {
	content: attr(data-on);
	left: 11px;
	color: #fff;
	text-shadow: 0 1px rgba(0, 0, 0, 0.2);
	opacity: 0;
}
.switch-input:checked ~ .switch-label {
	background: goldenrod;
	box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.15), inset 0 0 3px rgba(0, 0, 0, 0.2);
}
.switch-input:checked ~ .switch-label:before {
	opacity: 0;
}
.switch-input:checked ~ .switch-label:after {
	opacity: 1;
}
.switch-handle {
	position: absolute;
	top: 4px;
	left: 4px;
	width: 28px;
	height: 28px;
	background: linear-gradient(to bottom, #aaa 40%, #bbb);
	background-image: -webkit-linear-gradient(top, #aaa 40%, #bbb);
	border-radius: 100%;
	box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.2);
}
.switch-handle:before {
	content: "";
	position: absolute;
	top: 50%;
	left: 50%;
	margin: -6px 0 0 -6px;
	width: 12px;
	height: 12px;
	background: linear-gradient(to bottom, #aaa, #bbb);
	background-image: -webkit-linear-gradient(top, #aaa, #bbb);
	border-radius: 6px;
	box-shadow: inset 0 1px rgba(0, 0, 0, 0.02);
}
.switch-input:checked ~ .switch-handle {
	left: 74px;
	box-shadow: -1px 1px 5px rgba(0, 0, 0, 0.2);
}
 
/* Transition
========================== */
.switch-label, .switch-handle {
	transition: All 0.3s ease;
	-webkit-transition: All 0.3s ease;
	-moz-transition: All 0.3s ease;
	-o-transition: All 0.3s ease;
}