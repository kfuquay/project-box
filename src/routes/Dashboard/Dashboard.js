import React, { Fragment } from "react";
import ProjectsApiService from "../../services/project-api-service";
import ProjectContext from "../../Context/ProjectContext";
import ProjectTile from "../../components/ProjectTile/ProjectTile";
import LoadingIndicator from '../../components/LoadingIndicator/LoadingIndicator'
import "./Dashboard.css";

class Dashboard extends React.Component {
  static contextType = ProjectContext;

  constructor(props) {
    super(props);
    this.state = {
      isChecked: false,
      query: "",
      results: [],
      isLoading: false,
    };
  }

  componentDidMount() {
    this.context.clearError();
    this.setState({ isLoading: true })
    ProjectsApiService.getProjects()
      .then(this.context.setProjectList)
      .then(this.setState({isLoading: false}))
      .catch(this.context.setError);
  }

  handleSwitch = () => {
    this.setState({
      isChecked: !this.state.isChecked,
    });
  };

  updateQueryValue = e => {
    this.setState({ results: [], query: e.target.value }, function() {
      this.handleSearch();
    });
  };

  handleSearch = () => {
    const query = this.state.query;
    if (query.length >= 1) {
      const results = this.context.projectList.filter(project =>
        project.title.includes(`${query}`)
      );
      this.setState({ results });
    } else {
      const results = this.context.projectList;
      this.setState({ results });
    }
  };

  renderProjects() {
    const results =
      this.state.results.length < 1
        ? this.context.projectList
        : this.state.results;
    if (this.state.isChecked) {
      return results
        .filter(project => project.user_id === this.context.currentUserId)
        .map(project => <ProjectTile key={project.id} project={project} />);
    } else {
      return results.map(project => (
        <ProjectTile key={project.id} project={project} />
      ));
    }
  }

  render() {
    console.log(this.state.isLoading)
    const greeting =
      this.context.currentUser !== "" ? (
        <h2>{this.context.currentUser}'s Dashboard</h2>
      ) : (
        <h2>Dashboard</h2>
      );
    const { error } = this.context;
    return (
      <section className="project-list">
        <header className="list-header">{greeting}</header>
        {this.context.currentUser !== "" ? (
          <label className="switch">
            <input
              className="switch-input"
              type="checkbox"
              onChange={this.handleSwitch}
            />
            <span className="switch-label" data-on="Mine" data-off="All" />
            <span className="switch-handle" />
          </label>
        ) : (
          <Fragment />
        )}

        <div className="search-container">
          <input
            type="text"
            placeholder="search for..."
            value={this.state.query}
            onChange={e => this.updateQueryValue(e)}
          />
        </div>
        <div>
          {this.state.isLoading && this.context.error === null? <LoadingIndicator /> : <Fragment />}
        </div>

        <div className="list-projecttiles" aria-live="polite">
          {error ? (
            <p className="error">Sorry, there was an error</p>
          ) : (
            this.renderProjects()
          )}
        </div>
      </section>
    );
  }
}

export default Dashboard;
