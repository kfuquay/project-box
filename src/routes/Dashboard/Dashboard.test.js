import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dashboard from './Dashboard';

//TODO: write component test - uses context(clearError)... test currently returns TypeError: this.context.clearError is not a function..
describe.skip(`Dashboard  component`, () => {
  it("renders without crashing", () => {
    const context = {  clearError: () => {}
  }
    const wrapper = shallow(<Dashboard />, { context });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
