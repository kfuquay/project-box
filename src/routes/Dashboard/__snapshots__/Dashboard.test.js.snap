// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard  component renders without crashing 1`] = `
<section
  className="project-list"
>
  <header
    className="list-header"
  >
    <h2>
      's Dashboard
    </h2>
  </header>
  <label
    className="switch"
  >
    <input
      className="switch-input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="switch-label"
      data-off="All"
      data-on="Mine"
    />
    <span
      className="switch-handle"
    />
  </label>
  <div
    className="list-projecttiles"
  />
</section>
`;
